deepcave/runs/__init__.py

```diff
@@ -481,15 +486,17 @@ def get_config_id(self, config: Union[Configuration, Dict]) -> Optional[int]:
         Optional[int]
             The configuration id.
         """
+        # Convert the input configuration to a tuple
         if isinstance(config, Configuration):
             config = config.get_dictionary()
+        # Use same rounding as ConfigSpace does
+        input_config_tuple = config_to_tuple(config, ROUND_PLACES)
 
-        # Find out config id
-        for id, c in self.configs.items():
-            if c == config:
-                return id
-
-        return None
+        # Check if the input configuration tuple exists in the config id mapping
+        if input_config_tuple in self.config_id_mapping:
+            return self.config_id_mapping[input_config_tuple]
+        else:
+            return None
```

Comment:
Nice, this should be much faster when you have a lot of configs as it's a constant time lookup :)